sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/CommitLog.scala

```diff
@@ -72,12 +74,37 @@ class CommitLog(sparkSession: SparkSession, path: String)
 }
 
 object CommitLog {
-  private val VERSION = 1
+  private val VERSION = SQLConf.get.stateStoreCheckpointFormatVersion
```

Comment:
Why do we need to change this version?

Reply:
Currently we do not support switching back from checkpoint v2 to v1, the `validateVersion` function above ensures that.